### DISCUSSION TITLE
Welcome to OpenDDSharp Discussions

### DISCUSSION THREAD

**Discussion by jmmorato** [jmmorato/openddsharp#39] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bperumala** [comment]
Hi,

When i trying to communicate with OpenDDS-3.16 "DCPSInfoRepo" Service which is running in linux using window Sample Application
with OpenDDSSharp which is returning below
2021-03-31 17:55:08.591@<local_host>@2392@LM_ERROR@unused_arg_0|(2392|19120) ERROR: TopicImpl::enable, assert_topic failed with return value 7.
2021-03-31 17:55:08.591@<local_host>@2392@LM_DEBUG@unused_arg_0|(2392|19120) Service_Participant::get_discovery: returning repository for domain 42, repo DEFAULT_REPO.
2021-03-31 17:55:08.677@<local_host>@2392@LM_ERROR@unused_arg_0|(2392|19120) EXCEPTION, ERROR: InfoRepoDiscovery::assert_topic: 
system exception, ID 'IDL:omg.org/CORBA/MARSHAL:1.0'
Unknown vendor minor code id (0), minor code = 0, completed = NO

But when i try to with OpenDDS-3.14.1 "DCPSInfoRepo" Service which is running in linux, its working

When i observe your README file
OpenDDSharp has been compiled with OpenDDS v3.14

Can't we communicate with OpenDDS v3.16

Any help to compile OpenDDSSharp with OpenDDS v3.16

Thanks
Looking for help.

**Reply by jmmorato** [reply]
Hi,

I'm working on the `feature/a_pice_of_cake` branch to ease the build and release process. With the new build script it is easy to change the OpenDDS version and build your own version. If I recall correctly, `develop` branch is already build against 3.14.1 but not release yet.

Unfortunately, I have not too much time to work in OpenDDSharp due to the high load of work in my actual job. Anyway, I will try to finish soon that feature branch and release a new version using OpenDDS v3.16.0 in the meantime you can take a look by yourself and feel free to ask any question you have.

**Reply by bperumala** [reply]
HI,

Thank for Quick Response,
I am trying to run "feature/a_pice_of_cake" "OpenDDSharp.Build.CppCli.ps1" script but it is failing because "DDS_TAOv2.sln" file is not found as "configure" command is not getting executed.

And when is try to run "openddsharp-develop" Build "build_2019.cmd" getting long path problem.

Can i get cmd script 
to compile and generate openddssharp.dll against already downloaded openDDS-3.16 and compiled one

Thanks

![Capture131](https://user-images.githubusercontent.com/4962714/113193832-0ad1fa80-927e-11eb-9498-3f67f1749c15.PNG)

**Reply by jmmorato** [reply]
Did you run the ps script from `Build` folder?

It should download a new version of OpenDDS to the `ext` folder, you should not use your own version of OpenDDS, remove the OpenDDS environment variables if you added in your system.

I would recommend you to try first with OpenDDS v3.14.1 because the patches used in OpenDDSharp surely should be adapted for v3.16.0. 

I will try a build against v3.16.0 and give you better instructions.

**Comment by bperumala** [comment]
Hi,
When can we have OpenDDSSharp against OpenDDS-3.16.0 version? as i am trying to compile facing issue in my local environment

Waiting for release 

Thanks

**Reply by jmmorato** [reply]
There were some internal "breaking changes" for the migration to v3.16.

The `feature/a_piece_of_cake` branch now contain a migrated version to OpenDDS v3.16, no yet released but you can try to build it yourself.

**Comment by TheGrandJonsson** [comment]
Hi! 
When I'm trying to build the c++ idl project, the compilation fails on "TypeSupportImpl.h". In other words, it can't find the header file in the auto generated "TypeSupport.h" file. 
When searching manually for the file, it's located under the `/obj/Win32/Debug/` folder. But the auto generated file want it in the root folder. 
Am I missing something or is it a bug in the autogenreating code? I've just followed the guide on openddsharp webpage. 
Thanks!

**Reply by jmmorato** [reply]
Hi @TheGrandJonsson,

What is your selected solution platform? Perhaps Any CPU? If so, choose x86 or x64 to make it work correctly, otherwise, share your solution, if you can, and I will take a look to check if you have some issue in the setup.

PS: Please, open new discussion threads when asking for support. It will be easier for other user to find the information.

**Reply by TheGrandJonsson** [reply]
Hi! 
Can't really share my solution, but my platform is x86. I can however share the generated type support header if you want? 

I'll remember to create a new discussion thread in the future! 
Thanks!

**Reply by TheGrandJonsson** [reply]
@jmmorato I created a new project on the side and pushed it to Github, you can check the solution there. 

Thanks!

**Reply by jmmorato** [reply]
Ohh I see where the problem is...

I involuntarily released a new version of the IDL Visual Studio Extension, while working in the Continuous Delivery script,  that is not compatible with the current v0.5.0 version.

To fix the issue, you must replace in the `.vcxproj` all the occurrences of `$(IntDir)\$(ProjectName)` to just `$(IntDir)`. After that, force a `Rebuild` of the project (or `Clean` and then `Build`).

**Reply by TheGrandJonsson** [reply]
Yes, that was the issue!

Many thanks!
